File: algo/leetcode_283.py
''' Move all zeros to the end of the array
'''

def moveZeros(nums):
    if not nums: return
    i = 0; j = 0
    n = len(nums)
    for j, val in enumerate(nums):
        if 0 == val: 
            continue
        nums[i] = nums[j]
        i += 1
    while i < n:
        nums[i] = 0; i += 1


nums = [0, 1, 0, 3, 12]
moveZeros(nums)
            
        

File: algo/leetcode_283.cxx
#include <iostream>
#include <vector>

using namespace std;

void moveZeroes(vector<int> &nums) {
    int i = 0, j = i;
    for (; i < nums.size(); ++i) {
        if (0 != nums[i])
            nums[j++] = nums[i];
    }
    for (; j < nums.size(); ++j) {
        nums[j] = 0;
    }
}

int main() {
    auto nums = vector<int> {0, 1, 0, 3, 12};
    moveZeroes(nums);
    for (auto a : nums) cout << a << " "; cout << endl;
}
